src/sentry/models/integrations/sentry_app_installation.py

```diff
@@ -33,6 +36,22 @@ def get_installed_for_organization(self, organization_id: int) -> QuerySet:
     def get_by_api_token(self, token_id: str) -> QuerySet:
         return self.filter(status=SentryAppInstallationStatus.INSTALLED, api_token_id=token_id)
 
+    def get_projects(self, token: ApiToken) -> QuerySet[Project]:
+        from sentry.models import Project, SentryAppInstallationToken
+
+        try:
+            installation = self.get_by_api_token(token.id).first()
+        except SentryAppInstallation.DoesNotExist:
+            installation = None
+
+        if not installation:
+            return Project.objects.none()
+
+        if installation.sentry_app.status == SentryAppStatus.INTERNAL:
```

Comment:
@NisanthanNanthakumar you should definitely add a comment why were have to check this differently